addons/payment/models/payment_acquirer.py

```diff
@@ -336,6 +337,21 @@ def write(self, vals):
         self._check_required_if_provider()
         return result
 
+    def unlink(self):
+        """ Prevent the deletion of the payment acquirer if it has xmlid, because in that
+        case the record might be used in some modules.
+        """
+        external_ids = self.get_external_id()
+        for acquirer in self:
+            external_id = external_ids[acquirer.id]
+            if external_id \
+               and not external_id.startswith('__export__') \
+               and not self._context.get(MODULE_UNINSTALL_FLAG):
+                raise UserError(
+                    _("You cannot delete %s payment acquirer because it is used by the system.", acquirer.name)
+                )
+        return super(PaymentAcquirer, self).unlink()
```

Comment:
Simplify when `super` is called on `self`.
```suggestion
        return super().unlink()
```